tick/simulation/SCCS.py

```diff
@@ -0,0 +1,379 @@
+class SimuSCCS(Simu):
+    """Simulation of a Self Control Case Series (SCCS) model. This simulator can
+    produce exposure (features), outcomes (labels) and censoring data. 
+     
+    The features matrices  are a `n_sample` list of numpy arrays (dense case) or 
+    csr_matrices (sparse case) of shape `(n_intervals, n_features)` containing
+    exposures to each feature.
+    Exposure can take two forms:
+    - short repeated exposures: in that case, each column of the numpy arrays 
+    or csr matrices can contain multiple ones, each one representing an exposure
+    for a particular time bucket.
+    - infinite unique exposures: in that case, each column of the numpy arrays
+    or csr matrices can only contain a single one, corresponding to the starting
+    date of the exposure.
+
+    Parameters
+    ----------
+    n_samples : `int`,
+        Number of samples to generate.
+    
+    n_intervals : `int`
+        Number of time intervals used to generate features and outcomes.
+         
+    n_lags : `int`
+        Number of lags to be used when simulating the outcomes. The output
+        features are returned without lagging to allow the testing of lag
+        misspecification in the model.
+
+    sparse : `boolean`, default=True
+        Generate sparse or dense features.
+     
+    exposure_type : {'infinite', 'short'}, default='infinite'
+        Either 'infinite' for infinite unique exposures or 'short' for short
+        repeated exposures.
+        
+    distribution : {'multinomial', 'poisson'}, default='multinomial'
+        Dristribution used to generate the outcomes. In the 'multinomial' case,
+        the Poisson process used to generate the events is conditionned by total 
+        the number event per sample, which is set to be equal to one. In that
+        case, the simulation matches exactly the SCCS model hypotheses. In the
+        'poisson' case, the outcomes are generated from a Poisson process, which
+        can result in more than one outcome tick per sample. See 
+        `first_tick_only` option to filter out additional events.
+        
+    first_tick_only : `Boolean`, default=True
+        Keep only the first event and drop the consecutive ones when simulating
+        with Poisson distribution. 
+        
+    censoring : `Boolean`, default=True
+        Simulate a censoring vector. In that case, the features and outcomes are
+        simulated, then right-censored according to the simulated censoring
+        dates.
+        
+    censoring_prob : `float`, default=.7
+        Probability that a sample is censored. Should be in [0, 1].
+        
+    censoring_intensity : `float`, default=.9
+        The number of censored time intervals are drawn from a Poisson 
+        distribution with intensity equal to `censoring_intensity`. The higher,
+        the more intervals will be censored. Should be greater than 0.
+        
+    coeffs : `numpy.ndarray` of shape (n_features * (n_lags + 1)), default=None
```

Comment:
coeffs shape depends on n_lags and n_features, so it seemed weird to have it as the first argument. The default value of coeffs is None. In that case, random coefficients are generated. 
I can indeed add a function to generate more "structured" random coefficients